app/App/Http/Controllers/UserProfileController.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ProfileRequest;
+use App\Http\ViewModels\ProfileViewModel;
+use Domain\User\Actions\ChangePasswordAction;
+use Domain\User\Models\User;
+use Illuminate\Http\RedirectResponse;
+
+class UserProfileController
+{
+    public function index(User $user)
+    {
+        return (new ProfileViewModel($user))->view('userProfile.index');
+    }
+
+    public function update(
+        User $user,
+        ProfileRequest $request,
+        ChangePasswordAction $changePasswordAction
+    ): RedirectResponse {
+        $changePasswordAction($user, $request->input('password'));
```

Comment:
I think we should pass the hashed password to the action. I want to work with the hash as soon as possible, avoiding security issues in the future.

Say we'd make this action queuable (that's not likely to happen, I know); the plain password would be stored on the sever somewhere. We don't want that 😄

There's two ways we could approach this: hash the password in the controller, or create a DTO solely for passwords: `PasswordData`, which has a `$password_hash` property, and immediately hashes the password via a static constructor. For example: `PasswordData::fromPlainString($request->input('password'))`

What do you think?